src/main/java/org/prebid/server/floors/BasicPriceFloorAdjuster.java

```diff
@@ -22,10 +29,13 @@ public BigDecimal adjustForImp(Imp imp, String bidder, BidRequest bidRequest) {
             return imp.getBidfloor();
         }
 
-        // TODO: check with Bret 34 answer
-        final ImpMediaType mediaType = null;
+        final ImpMediaType mediaType = chooseMediaType(imp, extBidadjustmentfactors, bidder);
 
-        return resolveBidAdjustmentFactor(extBidadjustmentfactors, mediaType, bidder);
+        final BigDecimal factor = resolveBidAdjustmentFactor(extBidadjustmentfactors, mediaType, bidder);
+
+        return imp.getBidfloor() != null
```

Comment:
imp.getBidfloor() used three times in this method, pls introduce it to separate variable